### PR TITLE
Use exponential backoff for s3 503 error

### LINKED ISSUE #44: Handle S3 503 error
https://github.com/stoat-dev/stoat-action/actions/runs/3660320900/jobs/6187319317

### PR DESCRIPTION
**Pull request by tuliren**
Resolve #44.

### CODE REVIEW

File: action/__tests__/plugins/staticHosting/helpers.test.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+const mockCore = core as jest.Mocked<typeof core>;
+
+describe('uploadFileWithSignedUrl', function () {
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  describe('when there is no error', function () {
+    it('uploads the file with one fetch call', async () => {
+      mockFetch.mockResolvedValue({ ok: true } as Response);
```
Comment (tuliren) [inline action/__tests__/plugins/staticHosting/helpers.test.ts:20]: Should move this setup to a "beforeEach" block to be consistent across different tests within this file.

**Review by jrhizor — APPROVED** [review]
Looks good. 

Seems likely we'll want to use a library for this at some point like https://www.npmjs.com/package/p-retry as we have more operations that require similar retry logic.

**Comment by tuliren** [comment]
Good point. Will switch to use a library.

**Comment by tuliren** [comment]
I tried `p-retry`, but run into some ESM issues, and got this exception:

```
 FAIL  __tests__/plugins/staticHosting/plugin.test.ts
  ● Test suite failed to run

    Must use import to load ES Module: /Users/lirentu/git/stoat-action/action/node_modules/p-retry/index.js

       6 | // eslint-disable-next-line import/no-unresolved
       7 | import mime from 'mime-types';
    >  8 | import pRetry, { AbortError } from 'p-retry';
         | ^
       9 | import { basename, posix, resolve } from 'path';
      10 |
      11 | import { getApiUrlBase } from '../../stoatApiHelpers';

      at Runtime.requireModule (node_modules/jest-runtime/build/index.js:817:21)
      at Object.<anonymous> (src/plugins/staticHosting/helpers.ts:8:1)
      at Object.<anonymous> (src/plugins/staticHosting/plugin.ts:8:1)
      at Object.<anonymous> (src/plugins/staticHosting/index.ts:1:1)
      at Object.<anonymous> (__tests__/plugins/staticHosting/plugin.test.ts:4:1)
```

I'd rather not deal with this right now. Created an issue: #47.

**Pull request merged**